Web.md

```diff
@@ -299,6 +299,86 @@ Transcoding failure is detected by `decodeURIComponent`, which may throw
+source files (where a generated-code location consists of a line and column
+number, and a source location consists of file, line, and column). Generated
+files use a specially-formatted comment to specify a URL for a source map that
+describes the file.
+
+Source maps can be made to serve the same purpose for WebAssembly binary files,
+allowing parity with asm.js. By re-using an existing standard, browsers can
+take advantage of their existing source map support without having to make
+significant changes to their developer tools.
+
+
+### Alternatives
+
+An alternative method is to specify location mapping information in some way
+other than source maps. For example, it could be encoded directly into the WebAssembly
+binary the way function and local names are written in the "name" section. This
+is more convenient if the VM itself uses the information (for example to
+generate a stack trace). However existing developer tools implementations are
+designed around the source map model of passing a URL from the VM to the dev
+tools code, which fetches and interprets the information. Therefore it is
+expected that it will be much simpler to use "real" source maps than to design
+another format.
+
+Source maps also have known limitations (e.g. source variables cannot be
+described). A complete debug info format will eventually be needed; however the
+intention of this specification is to allow for easy integration with existing
+developer tools implementations in browsers.
+
+
+### Spec
+
+With few exceptions, most aspects of the source map
+[spec](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/)
+may be applied directly to WebAssembly.
+Because WebAssembly is a binary format (and thus does not have lines and columns, or
+comments per se), some simple reinterpretation of concepts is needed.
+
+#### Locations
+
+Source maps specify how to map a (zero-based) line and column position in
+generated code to a file, line, and column location in the source. For
+WebAssembly binary locations, the line number is always 1, and the column number
+is interpreted as a byte offset into the WebAssembly binary content (this results
+in a more efficient encoding than using the line number instead).
+Source locations are interpreted as in the source map spec.
+
+#### Linking generated code to source maps
+
+When the generated code is JavaScript, it includes a specially-formatted line
+at the end, which is the URL of the associated source map. For WebAssembly, a custom
+section named `"sourceMappingURL"` contains the URL.
+The URL is represented and encoded as defined in the WHATWG
+[URL spec](https://url.spec.whatwg.org).
+If the URL is not absolute, then:
+* If the wasm module is associated with an HTTP response (e.g. the
+compileStreaming or instantiateStreaming
+[APIs](#additional-web-embedding-api) are used), then it is resolved relative to
+the URL of the response.
+* Otherwise it is relative to the page's origin.
```

Comment:
Just to understand the relative resolution with "page's origin", if `WebAssembly.instantiate()` was called from a page's script at "http://example.org/~user/page.html", and wasm's `sourceMappingURL` is specified as "maps/t.wasm.map", the resulting source map URL will be "http://example.org/maps/t.wasm.map", is it correct? Does `document.baseURI` affect this?